src/payment/models/buy-crypto/services/buy-crypto.service.ts

```diff
@@ -52,18 +59,40 @@ export class BuyCryptoService {
     return await this.buyCryptoRepo.save(entity);
   }
 
+  async createFromCrypto(cryptoInputId: number, cryptoRouteId: number): Promise<BuyCrypto> {
+    let entity = await this.buyCryptoRepo.findOne({ cryptoInput: { id: cryptoInputId } });
+    if (entity) throw new ConflictException('There is already a buy crypto for the specified crypto input');
```

Comment:
I would remove this check, this is actually not possible and would be handled by the DB as a fallback.